File: 4.python/homeworks/caesar.py
import sys

if len(sys.argv) != 2 or not sys.argv[1].isdigit() or int(sys.argv[1]) < 0:
    print("Usage : python3 caesar.py positive_integer")
    exit(1) # c return equivalent

# to use cli argv : sys.argv[1]
key = int(sys.argv[1])

plain = input("plaintext : ")
cipher = ""

# convert char to int : ord()
# convert int to char : chr()
for i in range(len(plain)):
    # convert each char in ascii int, transform to new char with key
    # convert only char in azAZ
    if plain[i].isalpha():
        # check is lower or upper, then set the index
        if plain[i].islower():
            index = 97
        else:
            index = 65

        # set new char (%26 to cycle a-z) from index
        cipher_char = index + (ord(plain[i]) - index + key)%26
        # transform ascii int to char
        cipher_char = chr(cipher_char)
    else:
        cipher_char = plain[i]

    # concatenate in cipher
    cipher += cipher_char

print("ciphertext : " + cipher)

File: 1.c/homeworks/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

// accept cli argument for k
int main(int argc, string argv[])
{
    int index, i, k, start, n;
    string plaintext;

    if (argc != 2 )
    {
        printf("Usage: ./caesar k\nBad cli cipher key...Must quit now before you break everything...\n");
        return 1;
    }

    //transform string in int with atoi
    k = atoi(argv[1]);
    printf("%i\n", k);
    printf("plaintext: ");
    plaintext = get_string();
    n = strlen(plaintext);

    printf("ciphertext: ");

    for (i=0; i<n; i++)
    {
        // check if upper or lower
        if (isupper(plaintext[i]))
        {
            start = 65;
        } else if (islower(plaintext[i]))
        {
            start = 97;
        }

        if (isalnum(plaintext[i]))
        {
            // be sure that index is in range 0-26
            index = (plaintext[i] - start + k)%26;
            printf("%c", start + index);
        }
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");

    return 0;
}